### PR TITLE
*: add aliyun

### PR DESCRIPTION
**Pull request by arithx**

### CODE REVIEW

File: auth/aliyun.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+// ReadAliyunConfig decodes an aliyun config file
+//
+// If path is empty, $HOME/.aliyun/config.json is read.
+func ReadAliyunConfig(path string) (map[string]AliyunProfile, error) {
+	if path == "" {
+		user, err := user.Current()
+		if err != nil {
+			return nil, err
+		}
+		path = filepath.Join(user.HomeDir, OpenStackConfigPath)
```
Comment (jlebon) [inline auth/aliyun.go]: `AliyunConfigPath` ?

Comment (arithx) [reply auth/aliyun.go]: Whoops, yup. Nice catch!

**Comment by arithx** [comment]
Tested & marked ready for review

File: cmd/ore/aliyun/create.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Red Hat
```
Comment (ashcrow) [inline cmd/ore/aliyun/create.go]: Looking at other `cmd/ore/$CLOUD/`'s it seems like this file is generally called `create-image.go`, though there isn't total consensus:

```
$ find . -name "create.go"
./openstack/create.go
$ find . -name "create-image.go"
./azure/create-image.go
./gcloud/create-image.go
./do/create-image.go
```

File: cmd/ore/aliyun/create.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		fmt.Fprintf(os.Stderr, "Uploading image to object storage: %v\n", err)
+		os.Exit(1)
+	}
+
+	id, err := API.ImportImage(format, bucket, name, diskSize, device, name, description, architecture)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Couldn't create image: %v\n", err)
+		os.Exit(1)
+	}
+	fmt.Println(id)
```
Comment (ashcrow) [inline cmd/ore/aliyun/create.go]: Is there an output format expected here or is it free form? Examples I see:

## gcloud
```
        fmt.Printf("Creating image in GCE: %v...\n", imageNameGCE)
```

## do
Doesn't look like it outpus an id

File: network/ssh.go [hunk 4]
```diff
@@ -128,8 +128,9 @@ func ensurePortSuffix(host string, port int) string {
 
 func (a *SSHAgent) newClient(host string, user string, auth []ssh.AuthMethod) (*ssh.Client, error) {
 	sshcfg := ssh.ClientConfig{
-		User: user,
-		Auth: auth,
+		User:            user,
+		Auth:            auth,
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
```
Comment (ashcrow) [inline network/ssh.go]: Are we sure this is what we want here in all cases? https://godoc.org/golang.org/x/crypto/ssh#InsecureIgnoreHostKey

File: cmd/ore/aliyun/create.go (hunk 3 above)
Comment (arithx) [reply cmd/ore/aliyun/create.go]: There is not currently an expected output between different `ore` platforms. IMO there are two options here, we either leave it as is or I can update it to be a JSON output (similar to what is output by `copy-image`) in the format of `{'<region>': '<id>'}`.

**Pull request merged**